DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -3766,5 +3767,128 @@ IEnumerable<ulong> appliedTags
+
+            var bucket = this._rest.GetBucket(RestRequestMethod.GET, route, new { guild_id }, out var path);
+            var url = Utilities.GetApiUriFor(path);
+            var req = await this.DoRequestAsync(this._discord, bucket, url, RestRequestMethod.GET, route);
+            var rules = JsonConvert.DeserializeObject<IReadOnlyList<DiscordAutoModerationRule>>(req.Response);
+
+            return rules;
+        }
+
+        internal async Task<DiscordAutoModerationRule> ModifyGuildAutoModerationRuleAsync
+        (
+            ulong guild_id,
+            ulong rule_id,
+            Optional<string> name,
+            Optional<RuleEventType> event_type,
+            Optional<RuleTriggerMetadata> trigger_metadata,
+            Optional<IEnumerable<DiscordAutoModerationAction>> actions,
+            Optional<bool> enabled,
+            Optional<IEnumerable<ulong>> exempt_roles,
+            Optional<IEnumerable<ulong>> exempt_channels,
+            string reason = null
+        )
+        {
+            string route = $"{Endpoints.GUILDS}/:guild_id{Endpoints.AUTO_MODERATION}{Endpoints.RULES}/:rule_id";
+            var bucket = this._rest.GetBucket(RestRequestMethod.PATCH, route, new { guild_id, rule_id }, out var path);
+            var url = Utilities.GetApiUriFor(path);
+
+            var headers = Utilities.GetBaseHeaders();
+            if (!string.IsNullOrWhiteSpace(reason))
+                headers[REASON_HEADER_NAME] = reason;
+
+            string payload = DiscordJson.SerializeObject(new
+            {
+                name,
+                event_type,
+                trigger_metadata,
+                actions,
+                enabled,
+                exempt_roles = exempt_roles.HasValue ? exempt_roles.Value.ToArray() : Array.Empty<ulong>(),
+                exempt_channels = exempt_channels.HasValue ? exempt_channels.Value.ToArray() : Array.Empty<ulong>()
+            });
+
+            var req = await this.DoRequestAsync(this._discord, bucket, url, RestRequestMethod.PATCH, route, headers, payload);
+
+            return JsonConvert.DeserializeObject<DiscordAutoModerationRule>(req.Response);
+        }
+
+        internal async Task DeleteGuildAutoModerationRuleAsync(ulong guild_id, ulong rule_id, string reason)
+        {
+            string route = $"{Endpoints.GUILDS}/:guild_id{Endpoints.AUTO_MODERATION}{Endpoints.RULES}/:rule_id";
+
+            var bucket = this._rest.GetBucket(RestRequestMethod.DELETE, route, new { guild_id, rule_id }, out var path);
+            var url = Utilities.GetApiUriFor(path);
+
+            var headers = Utilities.GetBaseHeaders();
+            if (!string.IsNullOrWhiteSpace(reason))
+                headers[REASON_HEADER_NAME] = reason;
+
+            await this.DoRequestAsync(this._discord, bucket, url, RestRequestMethod.DELETE, route, headers);
+        }
+
+        internal async Task DeleteAllGuildAutoModerationRulesAsync(ulong guild_id, string reason)
```

Comment:
this shouldn't be part of the API client, imo, if it has no direct request matching it